EcoSoapBank/Payment/PaymentHistoryViewController.swift

```diff
@@ -0,0 +1,131 @@
+//
+//  PaymentHistoryViewController.swift
+//  EcoSoapBank
+//
+//  Created by Christopher Devito on 8/31/20.
+//  Copyright © 2020 Spencer Curtis. All rights reserved.
+//
+
+import UIKit
+
+class PaymentHistoryViewController: UIViewController {
+
+    var paymentController: PaymentController?
+    private var paymentCollectionView: UICollectionView = UICollectionView(frame: .zero, collectionViewLayout: UICollectionViewFlowLayout())
+    private var payments: [Payment] = [] {
+        didSet {
+            self.paymentCollectionView.reloadData()
+        }
+    }
+
+    var cellWidth: CGFloat {
+        paymentCollectionView.frame.size.width
+    }
+    var expandedHeight: CGFloat = 200
+    var notExpandedHeight: CGFloat = 80
+    var isExpanded = [Bool]()
+    let cellIdentifier = "PaymentCell"
+
+    override func loadView() {
+        view = BackgroundView()
+    }
+
+    override func viewWillLayoutSubviews() {
+        let width = self.view.frame.width
+        let navigationBar: UINavigationBar = UINavigationBar(frame: CGRect(origin: CGPoint(x: 0, y: 0), size: CGSize(width: width, height: 44)))
+        view.addSubview(navigationBar)
+        let navBarTitle = UINavigationItem(title: "Payment History")
+        navigationBar.setItems([navBarTitle], animated: false)
+    }
```

Comment:
Instead of this, this vc should be embedded in a navigation controller. This will get called multiple times, adding a navigation bar each time. I know you aren't navigating from this view (neither is my impact vc) but embedding in a navigation controller does not hurt anything.

Reply:
Ah, ok. I actually don't know how to do that, but I'll look into it.